### ISSUE TITLE
Not compatible with elasticsearch>=2.0.0

### ISSUE THREAD

**Issue by miltontony** [universalcore/elastic-git#65]

**Comment by miltontony** [comment]
Issue created with elasticutils https://github.com/mozilla/elasticutils/issues/292

**Comment by infojunkie** [comment]
We'd love to try out elastic-git, but compatibility with latest ES is pretty important for our infrastructure. Is upgrading the codebase on your roadmap?

**Comment by smn** [comment]
@infojunkie we'd love see it upgraded but it isn't a huge priority on our roadmap. The biggest issue is that Mozilla's elasticutils isn't being maintained anymore and that's a big dependency for us.

The work involved here is either taking over maintenance of elasticutils or swapping it out for an ES library that is maintained while (hopefully) maintaining backwards compatibility to some degree with our existing API. The latter is going to be quite difficult as our API relies heavily on elasticutils' API.

**Comment by infojunkie** [comment]
Yes, I noticed the tight coupling in the code. As far as ES python libs go, I would suggest Elastic's own `elasticsearch-py` as a base (https://github.com/elastic/elasticsearch-py), since it has good likelihood to remain maintained. Regarding minimizing rework, one approach could involve recreating the `elasticutils` API in an intermediate classlib, using the lower-level `elasticsearch-py`.

**Comment by infojunkie** [comment]
As we get more familiar with `elastic-git`, we might decide to use it in production and devote resources to the upgrade. We'll keep you updated here.

**Comment by smn** [comment]
That'd be awesome!

Yes, it's entirely possible. Other than our own team's voluntary contributions, we have no projects currently driving this codebase and so it's not really possible for us to commit to it.

**Comment by webmaven** [comment]
Is this still an active issue?

**Comment by miltontony** [comment]
@webmaven yes it is, we haven't upgraded it yet. But we're still using it in production with ES `1.7`

**Comment by webmaven** [comment]
How about now? :grinning:

**Comment by smn** [comment]
@webmaven we're very happy to accept code contributions but have no capacity to drive this ourselves and will likely not be a priority for us in the future either.

**Comment by jamessandy** [comment]
I want to contribute @webmaven
